src/ring_theory/valuation/valuation_subring.lean

```diff
@@ -300,6 +300,80 @@ instance linear_order_overring : linear_order { S | A ≤ S } :=
   decidable_le := infer_instance,
   ..(infer_instance : partial_order _) }
 
+/-- A directed union of valuation subrings is a valuation subring.
+NOTE: The carrier of this valuation subring is defeq to the union of the family. -/
+def supr_of_directed {ι : Type*} [nonempty ι]
+  (f : ι → valuation_subring K) (hf : directed (≤) f) :
+  valuation_subring K :=
+{ mem_or_inv_mem' := begin
+    intros x,
+    inhabit ι,
+    cases (f default).mem_or_inv_mem x,
+    { left, refine set.mem_Union_of_mem _ h },
+    { right, refine set.mem_Union_of_mem _ h }
+  end,
+  ..(subring.copy (⨆ (i : ι), (f i).to_subring)
+      (⋃ (i : ι), (f i : set K)) (subring.coe_supr_of_directed hf).symm) }
+
+lemma mem_supr_of_directed {ι : Type*} [nonempty ι]
+  (f : ι → valuation_subring K) (hf : directed (≤) f) (x : K) :
+  x ∈ supr_of_directed f hf ↔ ∃ i : ι, x ∈ f i :=
+set.mem_Union
+
+lemma coe_supr_of_directed {ι : Type*} [nonempty ι]
+  (f : ι → valuation_subring K) (hf : directed (≤) f) :
+  (supr_of_directed f hf : set K) = ⋃ (i : ι), f i := rfl
+
+lemma le_supr_of_directed {ι : Type*} [nonempty ι]
+  (f : ι → valuation_subring K) (hf : directed (≤) f) (i : ι) :
+  f i ≤ supr_of_directed f hf :=
+set.subset_Union _ i
+
+lemma supr_of_directed_le {ι : Type*} [nonempty ι]
+  (f : ι → valuation_subring K) (hf : directed (≤) f) (A : valuation_subring K)
+  (h : ∀ i, f i ≤ A) :
+  supr_of_directed f hf ≤ A :=
+set.Union_subset h
```

Comment:
Note that in addition to the new `variables` I suggest above, the context already contains `(A : valuation_subring K)` so this can be just:
```suggestion
lemma supr_of_directed_le (h : ∀ i, f i ≤ A) : supr_of_directed f hf ≤ A := set.Union_subset h
```

Reply:
and likewise below